lib/r10k/action/deploy/environment.rb

```diff
@@ -112,6 +103,10 @@ def write_environment_info!(environment, started_at)
             f.puts(JSON.pretty_generate(deploy_info))
           end
         end
+
+        def allowed_initialize_opts
+          super.merge(puppetfile: true, cachedir: true)
```

Comment:
`:purge` is a bit strange in that I meant to make it a tunable and the old code supported it, but I never got around to passing through the option from the CLI to the Action. Good catch, I'll add this.